src/plugins/kibana_react/public/app_links/redirect_cross_app_link.test.tsx

```diff
@@ -0,0 +1,266 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { MouseEvent } from 'react';
+import { mount } from 'enzyme';
+import { applicationServiceMock } from '../../../../core/public/mocks';
+import { RedirectCrossAppLinks } from './redirect_cross_app_link';
+import { BehaviorSubject } from 'rxjs';
+
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+
+describe('RedirectCrossAppLinks', () => {
+  let application: ReturnType<typeof applicationServiceMock.createStartContract>;
+
+  beforeEach(() => {
+    application = applicationServiceMock.createStartContract();
+    application.parseAppUrl.mockReturnValue({ app: 'targetApp' });
+    application.currentAppId$ = new BehaviorSubject<string>('currentApp');
+  });
+
+  it('intercept click events on children link elements', () => {
+    let event: MouseEvent;
+
+    const component = mount(
+      <div
+        onClick={(e) => {
+          event = e;
+        }}
+      >
```

Comment:
I feel like these enzyme based tests are as good as actual FTR test, as the only difference is that they are using jsdoc instead of the actual browser implementation. It also allow more precise testing, as we can collect the event to perform verifications,  which would not be possible on FTR tests.

Tell me if you think that adding FTR tests is still necessary.

Reply:
Nope. Seems fine to me. Can be covered by FTR when in use by actual applications.